### DISCUSSION TITLE
What API methods to simulate multiple different plans simultaneously

### DISCUSSION THREAD

**Discussion by fatensj** [USC-ACTLab/crazyswarm#470] [Q&A]
Hi,
I am working on multi UAV path planning algorithm (simulation), and I would like to have the waypoints in a CSV file such that:
UAV#1
[x0,y0,z0]....[xn,yn,n]
.
.
. 
UAV#n
[x0,y0,z0]....[xn,yn,n]

Then I want to know how to generate the trajectories at first and then display them simultaneously on the 3D plane.
Thank you

**Comment by whoenig** [comment]
Motion planning: If you are fine with stopping at each waypoint, you can use the goTo command, see https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/scripts/waypoints.py. Otherwise, you will need to use a trajectory planner to precompute trajectories (e.g., https://doi.org/10.1109/TRO.2018.2853613), and execute them, see https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/scripts/swap6v.py.

Display: you can use the `--sim` flag to see a visualization. There is currently no tool that plots a static trajectory (which IMO would not be that useful, as collisions won't be apparent).

**Reply by fatensj** [reply]
Thank you for the answer. I tested the examples you suggested.
For now, the code in waypoints.py seems sufficient. I also visualized the paths, and it came pretty well. It just was fast abit.

However, I've tried to execute the example in swap6v.py, but I got an error. I think that there are requirements to execute this example? Do I need to save the trajectories in a file first?.
I also looked into code to generate trajectories. Can I use this https://github.com/whoenig/uav_trajectories

Thank you

**Reply by whoenig** [reply]
swap6v: You need to have robots with certain IDs, as defined here: https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/scripts/swap6v.py#L27. Of course, you can also change the IDs in the script.

You can use the uav_trajectories repo to generate smooth trajectories for a single robot. There is no inter-robot collision-checking (for sparse scenarios, you may use the on-board collision avoidance mode).
